From 520acbb8d8ebfb1c301eccdfa26523c8fe2cb9e0 Mon Sep 17 00:00:00 2001
From: rramasamy <rramasamy@maxlinear.com>
Date: Mon, 26 May 2025 20:31:37 +0800
Subject: [PATCH] agent: ap_autoconfiguration_task: agent sends wrong center
 frequency

During Easymesh onboarding, in SLAVE_JOINED_NOTIFICATION the center frequency
for 6GHz with 160MHz was sent wrong.
Instead of sending center frequency, Agent is sending primary_80MHz center
frequency for 6GHz in 160MHz.

It should be m_center_frequency_2 and not m_center_frequency for 6GHz if the
BW is 160MHz.

Closes PPM-3325

Signed-off-by: rramasamy <rramasamy@maxlinear.com>
---
 .../beerocks/slave/tasks/ap_autoconfiguration_task.cpp | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp b/agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp
index 03c87a4b31..47bedc3fe9 100644
--- a/agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp
+++ b/agent/src/beerocks/slave/tasks/ap_autoconfiguration_task.cpp
@@ -884,8 +884,14 @@ bool ApAutoConfigurationTask::send_ap_autoconfiguration_wsc_m1_message(
     notification->cs_params().bandwidth = radio->wifi_channel.get_bandwidth();
     notification->cs_params().channel_ext_above_primary =
         radio->wifi_channel.get_ext_above_primary();
-    notification->cs_params().vht_center_frequency = radio->wifi_channel.get_center_frequency();
-    notification->cs_params().tx_power             = radio->tx_power_dB;
+    if (radio->wifi_channel.get_freq_type() == eFreqType::FREQ_6G &&
+        radio->wifi_channel.get_bandwidth() == eWiFiBandwidth::BANDWIDTH_160) {
+        notification->cs_params().vht_center_frequency =
+            radio->wifi_channel.get_center_frequency_2();
+    } else {
+        notification->cs_params().vht_center_frequency = radio->wifi_channel.get_center_frequency();
+    }
+    notification->cs_params().tx_power = radio->tx_power_dB;
 
     // The add_vs_tlv method invokes the handler to add Vendor specific TLVs to the
     // AP_AUTOCONFIGURATION_WSC_MESSAGE.
-- 
GitLab

